### ISSUE TITLE
getting no checkpoints while training (training is too slow?)

### ISSUE THREAD

**Issue by byuns9334** [NVIDIA/waveglow#151]
Hi, thanks for your great work, I am training waveglow with 3 gpus, but I still don't get any checkpoints. The "iters_per_checkpoint", "epochs" parameters are 1, 100000, respectively. The training data size is only about 2G, definitely smaller than LJSpeech. And the batch size is 8. All gpus are 100% running. I'm using `distributed.py` currently.  I believe I installed `apex`.

This is the only log message I got :

![image](https://user-images.githubusercontent.com/26813646/63820907-051b8100-c986-11e9-94eb-879a6eb537a7.png)

How can I make distributed.py actually work? 

This is the config I'm using: 

![image](https://user-images.githubusercontent.com/26813646/63821009-59befc00-c986-11e9-8e1f-4cbef9ae5ae4.png)

Thank you so much!

**Comment by jongwook** [comment]
1. Does it make checkpoints if you run the same command with `train.py` instead?
2. Is `torch.distributed.is_nccl_available()` true?
3. What does the stacktrace look like if you press `Ctrl+C` when the program seems to be hanging?

**Comment by rafaelvalle** [comment]
@byuns9334 can you see training loss values in our stdout?

**Comment by rafaelvalle** [comment]
Closing due to inactivity.

**Comment by crackedeggs1** [comment]
Having the same problem. iters_per_checkpoint: 10, epochs: 400 (while testing this problem), but absolutely no checkpoints are being saved in the output_directory. Small data set for testing (about 10 WAV files) and only using 1 GPU w/ train.py

Is there a specific config.json key that I need to set to true in order to enable checkpointing? It's kind of pointless to train if it will just forget everything once the run is completed.
